stream.go

```diff
@@ -206,6 +201,8 @@ func (st *Stream) produceKVs(ctx context.Context, threadId int) error {
 	}
 	defer txn.Discard()
 
+	outList := new(pb.KVList)
+	outList.AllocRef = st.newAllocator(threadId).Ref
```

Comment:
Why do we want to move it outside iterate? I see no difference with case when if it is inside `iterate`.

Reply:
Also, do we want a test where `KeyToList()` does not use the allocator?

Reply:
The outlist was inside `iterate` function and `sendit` creates a new outlist.
So if the following sequence of calls happen
```
iterate <- New outlist created (say o1)
sendIt <- Old outlist (o1) pushed to kvchan and new outlist (say o2) created 
iterate <- new outlist (o3 created)
```
Notice that we did not push 02 to the `kvChan` and so the allocator in `o2` will leak.
Moving the outlist outside prevents this problem.

Reply:
I'll push a test for this.